<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 üì• Load data into R environment | tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package</title>
  <meta name="description" content="Chapter 1 üì• Load data into R environment | tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 üì• Load data into R environment | tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 üì• Load data into R environment | tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package" />
  
  
  

<meta name="author" content="Yihao Chen, Li-Na He, Yuanzhe Zhang, Jingru Gong, Shuangbin Xu, Yuelong Shu, Di Zhang, Guangchuang Yu, Zhixiang Zuo" />


<meta name="date" content="2024-06-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="batch-effect-correction.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">tigeR documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>üìñ <strong>Getting start with tigeR</strong></a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>0.1</b> üî¨ Introduction</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#the-workflow-of-tiger"><i class="fa fa-check"></i><b>0.2</b> üèû The workflow of tigeR</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>0.3</b> üõ†Ô∏è Installation</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#function-overview"><i class="fa fa-check"></i><b>0.4</b> üó∫Ô∏è Function overview</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#troubleshooting"><i class="fa fa-check"></i><b>0.5</b> üí• Troubleshooting</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html"><i class="fa fa-check"></i><b>1</b> üì• Load data into R environment</a>
<ul>
<li class="chapter" data-level="1.1" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html#organization-of-built-in-data"><i class="fa fa-check"></i><b>1.1</b> Organization of built-in data</a></li>
<li class="chapter" data-level="1.2" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html#obtain-data-from-tiger.data-package-recommended"><i class="fa fa-check"></i><b>1.2</b> Obtain data from tigeR.data package (recommended)</a></li>
<li class="chapter" data-level="1.3" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html#obtain-data-from-tiger-web-server"><i class="fa fa-check"></i><b>1.3</b> Obtain data from tigeR web server</a></li>
<li class="chapter" data-level="1.4" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html#obtain-data-from-experimenthub"><i class="fa fa-check"></i><b>1.4</b> Obtain data from ExperimentHub</a></li>
<li class="chapter" data-level="1.5" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html#pre-processing-of-custom-data"><i class="fa fa-check"></i><b>1.5</b> Pre-processing of custom Data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="batch-effect-correction.html"><a href="batch-effect-correction.html"><i class="fa fa-check"></i><b>2</b> üñéüèª Batch Effect Correction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="batch-effect-correction.html"><a href="batch-effect-correction.html#batch-correction-method"><i class="fa fa-check"></i><b>2.1</b> Batch Correction Method</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html"><i class="fa fa-check"></i><b>3</b> üö© Biomarker Evaluation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#integrate-analysis"><i class="fa fa-check"></i><b>3.1</b> Integrate analysis</a></li>
<li class="chapter" data-level="3.2" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#differential-analysis"><i class="fa fa-check"></i><b>3.2</b> Differential analysis</a></li>
<li class="chapter" data-level="3.3" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#suvival-analysis"><i class="fa fa-check"></i><b>3.3</b> Suvival analysis</a></li>
<li class="chapter" data-level="3.4" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#calculate-comprehensive-signature-score"><i class="fa fa-check"></i><b>3.4</b> Calculate comprehensive signature score</a></li>
<li class="chapter" data-level="3.5" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#assess-the-performance-of-signature"><i class="fa fa-check"></i><b>3.5</b> Assess the performance of signature</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="tumor-microenvironment-analysis.html"><a href="tumor-microenvironment-analysis.html"><i class="fa fa-check"></i><b>4</b> üå± Tumor Microenvironment Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="tumor-microenvironment-analysis.html"><a href="tumor-microenvironment-analysis.html#availiable-tme-analysis-method-in-tiger"><i class="fa fa-check"></i><b>4.1</b> Availiable TME analysis method in tigeR</a></li>
<li class="chapter" data-level="4.2" data-path="tumor-microenvironment-analysis.html"><a href="tumor-microenvironment-analysis.html#derive-the-proportions-of-different-tme-cell-types"><i class="fa fa-check"></i><b>4.2</b> Derive the proportions of different TME cell types</a></li>
<li class="chapter" data-level="4.3" data-path="tumor-microenvironment-analysis.html"><a href="tumor-microenvironment-analysis.html#visualization-and-comparing-the-cell-proportions"><i class="fa fa-check"></i><b>4.3</b> Visualization and comparing the cell proportions</a></li>
<li class="chapter" data-level="4.4" data-path="tumor-microenvironment-analysis.html"><a href="tumor-microenvironment-analysis.html#searching-for-key-cell-types-associated-with-immunotherapy-response"><i class="fa fa-check"></i><b>4.4</b> Searching for key cell types associated with immunotherapy response</a></li>
<li class="chapter" data-level="4.5" data-path="tumor-microenvironment-analysis.html"><a href="tumor-microenvironment-analysis.html#generate-a-custom-reference-matrix-for-tme-deconvolution-from-single-cell-sequencing-data"><i class="fa fa-check"></i><b>4.5</b> Generate a custom reference matrix for TME deconvolution from single-cell sequencing data</a></li>
<li class="chapter" data-level="" data-path="tumor-microenvironment-analysis.html"><a href="tumor-microenvironment-analysis.html#more-details-about-tme-analysis"><i class="fa fa-check"></i>üìù More details about TME analysis</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="prediction-model-construction.html"><a href="prediction-model-construction.html"><i class="fa fa-check"></i><b>5</b> ü§ñ Prediction Model Construction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="prediction-model-construction.html"><a href="prediction-model-construction.html#build-model"><i class="fa fa-check"></i><b>5.1</b> Build Model</a></li>
<li class="chapter" data-level="5.2" data-path="prediction-model-construction.html"><a href="prediction-model-construction.html#test-model"><i class="fa fa-check"></i><b>5.2</b> Test Model</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="response-prediction.html"><a href="response-prediction.html"><i class="fa fa-check"></i><b>6</b> üîÆ Response Prediction</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="load-data-into-r-environment" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> üì• Load data into R environment<a href="#load-data-into-r-environment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="organization-of-built-in-data" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Organization of built-in data<a href="load-data-into-r-environment.html#organization-of-built-in-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The built-in data follows several specifications for constructing:</p>
<p>‚ë† They are constructed into a SummarizedExperiment object using the <strong>SummarizedExperiment()</strong> function in SummarizedExperiment package.</p>
<p>‚ë° The assays of the SummarizedExperiment object are stored as a SimpleList containing an expression matrix, where rows represent genes and columns represent samples.</p>
<p>‚ë¢ The colData of the SummarizedExperiment object is a table(DFrame object of S4), where the row names must be the same as the column names of the expression matrix. The column names of the table represent the clinical information of patients. The detailed information about the column names is presented below.</p>
<div style="width:780px; height:400px; overflow-y: scroll; overflow-x: hidden;">
<table>
<colgroup>
<col width="17%" />
<col width="17%" />
<col width="47%" />
<col width="17%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Column name</th>
<th align="center">Recommend value</th>
<th align="center">Description</th>
<th align="center">Necessity in tigeR analysis</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">sample id</td>
<td align="center">character</td>
<td align="center">The sample identity.</td>
<td align="center">Required</td>
</tr>
<tr class="even">
<td align="center">patient_name</td>
<td align="center">character</td>
<td align="center">The patient identity.</td>
<td align="center">Alternative</td>
</tr>
<tr class="odd">
<td align="center">dataset_id</td>
<td align="center">character</td>
<td align="center">The dataset identity.</td>
<td align="center">Alternative</td>
</tr>
<tr class="even">
<td align="center">Treatment</td>
<td align="center">PRE/POST</td>
<td align="center">The patient sample is collected after Treatment or before Treatment</td>
<td align="center">Required</td>
</tr>
<tr class="odd">
<td align="center">response</td>
<td align="center">CR/MR/PR/PD/SD/NR/R/N</td>
<td align="center">The immunotherapy response.</td>
<td align="center">Alternative</td>
</tr>
<tr class="even">
<td align="center">response_NR</td>
<td align="center">R/N</td>
<td align="center">The immunotherapy response which only contains R or N.</td>
<td align="center">Required</td>
</tr>
<tr class="odd">
<td align="center">M.stage</td>
<td align="center">character</td>
<td align="center">The M stage of patients.</td>
<td align="center">Alternative</td>
</tr>
<tr class="even">
<td align="center">overall.survival..days.</td>
<td align="center">value</td>
<td align="center">The number of days, months or years of survival time of patients.(all samples must be the same unit)</td>
<td align="center">Required (for survival analysis)</td>
</tr>
<tr class="odd">
<td align="center">vital.status</td>
<td align="center">Alive/Dead</td>
<td align="center">The survival status of patient.</td>
<td align="center">Required (for survival analysis)</td>
</tr>
<tr class="even">
<td align="center">Total.Mutation</td>
<td align="center">numeric</td>
<td align="center">The total mutation gene numbers.</td>
<td align="center">Alternative</td>
</tr>
<tr class="odd">
<td align="center">Gender</td>
<td align="center">M/F</td>
<td align="center">The gender of patient.</td>
<td align="center">Alternative</td>
</tr>
<tr class="even">
<td align="center">Therapy</td>
<td align="center">character</td>
<td align="center">The anti-tumor therapy on patients.</td>
<td align="center">Alternative</td>
</tr>
<tr class="odd">
<td align="center">age_start</td>
<td align="center">numeric</td>
<td align="center">The age of the patient at diagnosis.</td>
<td align="center">Alternative</td>
</tr>
<tr class="even">
<td align="center">tumor_type</td>
<td align="center">character</td>
<td align="center">The type of tumor.</td>
<td align="center">Alternative</td>
</tr>
<tr class="odd">
<td align="center">seq_type</td>
<td align="center">character</td>
<td align="center">The sequencing type.</td>
<td align="center">Alternative</td>
</tr>
<tr class="even">
<td align="center">id</td>
<td align="center">character</td>
<td align="center">The identity of dataset.</td>
<td align="center">Alternative</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="obtain-data-from-tiger.data-package-recommended" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Obtain data from tigeR.data package (recommended)<a href="load-data-into-r-environment.html#obtain-data-from-tiger.data-package-recommended" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="load-data-into-r-environment.html#cb2-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;YuLab-SMU/tigeR.data&quot;</span>) </span>
<span id="cb2-2"><a href="load-data-into-r-environment.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tigeR.data) <span class="do">## load all built-in data set into R environment</span></span></code></pre></div>
</div>
<div id="obtain-data-from-tiger-web-server" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Obtain data from tigeR web server<a href="load-data-into-r-environment.html#obtain-data-from-tiger-web-server" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="load-data-into-r-environment.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Dataloader</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">use_source=</span><span class="st">&quot;Web Server&quot;</span>)</span></code></pre></div>
</div>
<div id="obtain-data-from-experimenthub" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Obtain data from ExperimentHub<a href="load-data-into-r-environment.html#obtain-data-from-experimenthub" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="load-data-into-r-environment.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Dataloader</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">use_source=</span><span class="st">&quot;ExperimentHub&quot;</span>)</span></code></pre></div>
</div>
<div id="pre-processing-of-custom-data" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> Pre-processing of custom Data<a href="load-data-into-r-environment.html#pre-processing-of-custom-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>‚ÄÉWhen conducting analysis using custom data, you need to pre-process your data and construct a SummarizedExperiment object. Prepare at least one gene expression matrix (rows for genes, columns for samples) and a data frame including corresponding clinical information of the samples. The data frame should include at least the following information:</p>
<ul>
<li>sample id</li>
<li>Treatment (PRE/POST)</li>
<li>response_NR (R/N)</li>
<li>overall.survival..days. (for survival analysis)</li>
<li>vital.status (for survival analysis)</li>
</ul>
<p>‚ÄÉHere is a brief example:</p>
<div style="width:780px; height:300px; overflow-y: scroll; overflow-x: hidden;">
<pre><code>## Construt FPKM Matrix
genes &lt;- c(&quot;5S_rRNA&quot;, &quot;A1BG&quot;, &quot;A1BG-AS1&quot;)
samples &lt;- c(&quot;SRR3184279&quot;, &quot;SRR3184280&quot;, &quot;SRR3184281&quot;)

mtr &lt;- matrix(c(0.04115157, 0.01311394, 0.02275499,
                0.07774626, 0.29494916, 0.11743900,
                2.01782650, 0.83763027, 0.77378394),
              nrow = 3, ncol = 3, byrow = TRUE, 
              dimnames = list(genes, samples))

## Construct Clinical Information Table
data &lt;- data.frame(
  sample_id = c(&quot;SRR3184279&quot;, &quot;SRR3184280&quot;, &quot;SRR3184281&quot;),
  Treatment = c(&quot;PRE&quot;, &quot;PRE&quot;, &quot;PRE&quot;),
  response_NR = c(&quot;N&quot;, &quot;R&quot;, &quot;R&quot;),
  overall_survival_days = c(607, 927, 948),
  vital_status = c(&quot;Dead&quot;, &quot;Alive&quot;, &quot;Alive&quot;),
  stringsAsFactors = FALSE)

## Construct SummarizedExperiment Object
SE &lt;- SummarizedExperiment::SummarizedExperiment(assays = list(fpkm = mtr),
                                                 colData = data)

SE
# class: SummarizedExperiment
# dim: 3 3
# metadata(0):
#   assays(1): fpkm
# rownames(3): 5S_rRNA A1BG A1BG-AS1
# rowData names(0):
#   colnames(3): SRR3184279 SRR3184280 SRR3184281
# colData names(5): sample_id Treatment response_NR overall_survival_days
# vital_status</code></pre>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="batch-effect-correction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-Load-data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
