---
title: "tigeR Vignette"
author: "Guangchuang Yu"
output: rmarkdown::html_vignette
prettydoc::html_pretty:
    toc: false
    theme: cayman
    highlight: github
vignette: >
  %\VignetteIndexEntry{tigeR}
  %\usepackage[utf8]{inputenc}
  %\VignetteEngine{knitr::rmarkdown}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(tigeR)
```

# 1 Signature Discovery
## 1.1 Gini_rank()
&emsp;You can use ***Gini_rank()*** function to calculate the Gini index of different genes in a SummarizedExperiment objects. The Gini index of each gene in data set ***D*** is calculated by the following formula: 
  
&emsp;___Gini_index(D,A) = (|D^H^| / |D|) * Gini(D^H^) +  (|D^L^| / |D|) * Gini(D^L^)___
  
&emsp;where ***D, D^R^, D^N^*** represents the data set ***D***. The samples which expression of gene **A** is higher or equal to the median value in the data set, and the samples which expression of gene **A** is lower to the median value in the data set. ***|D|, |D^H^|, |D^L^|*** represents sample size of them.  
  
  
Gini values of ***D^R^*** and ***D^N^*** are calculated by the following formula:  
  
&emsp;___Gini(D^H^) = (|D^H^| - |D^HR^|^2^ - |D^HN^|^2^) / |D^H^|___ 
 
&emsp;___Gini(D^L^) = &nbsp;(|D^L^| - |D^LR^|^2^ - |D^LN^|^2^) / |D^L^|___
    
&emsp;where ***D^HR^***, ***D^HN^*** represent Responder in samples which expression of gene **A** is higher than median value, Non-Responder in samples which expression of gene **A** is higher than median value and ***D^LR^***, ***D^LN^*** as well.
  
  
&emsp;The Gini index indicates the misclassified probability of a sample which randomly draw from a sample set. **A** gene which has low Gini index always contains more classification information and is more suitable as a feature of a machine learning model.

```         
Gini_rank(MEL_GSE78220)
```

## 1.2 diff_gene()

&emsp;Return differential expression genes which |log(FC)| > 1.5 and the P value < 0.05.

```         
diff_gene(SE = MEL_GSE78220, threshold = c(1.5, 0.05))
```

# 2 Response Prediction

## 2.1 Predict Response using existing Signatures

&emsp;By using **predict_Signature()** function, you can get a overall Signature score matrix of 23 Signatures in tigeR which collumns are Signature score and rows are sample names.

```         
result <- predict_Signature(MEL_GSE78220)
```

## 2.2 Predict Response using built-in functions in tigeR

&emsp;You can use function **build_Model** to build predictive models including naive Bayes models, Support vector machine model, random forest model, Cancerclass model, Adaboost model, Logitboost model, Logistics regression model.

&emsp;Parameter *SE* is a SummarizedExperiment object or a list contains SummarizedExperiment objects (including expression profile and patient clinical information).

&emsp;Parameter *Signature* is a subset of genes which you are interested in.

&emsp;Parameter *_rmBE_* is a logical variable. If TRUE, the function will remove the batch effect between datasets through the built-in batch effect removal tool *Combat*.

&emsp;Parameter *response_NR* is a logical variable.If TRUE, the function will automatically convert the patient's drug response (such as PR, NR, SD, etc. to binary value NR (non-responder) or R (Responder)).

The function returns a trained model. By applying **predict()** fuction, users can .

See details in <https:://github/YuLab-SMU/tigeR>

# 3 Tumor Microenviroment Deconvolution

## 3.1 CIBERSORT()

CIBERSORT is an analytical tool from the Alizadeh Lab and Newman Lab to impute gene expression profiles and provide an estimation of the abundances of member cell types in a mixed cell population, using gene expression data.

tigeR offer you an built-in function CIBERSORT() for estimation of the abundances of member cell types.

```    
result <- CIBERSORT(sig_matrix = LM22, SE = MEL_GSE78220, perm = 0, QN = TRUE)
result[[2]]
```

parameter sig_matrix is a expression matrix of Signature genes from each type of immune cells LM22 is a matrix which contain expression of Signature genes of each type of immune cell

parameter mix_matrix is the expression matrix of bulk RNA-seq which you want to know its cell fraction

parameter perm is the number of permutations

parameter QN perform quantile normalization or not (TRUE/FALSE)

CIBERSORT function will return a list which first element is the cell type proportion matrix and second element is a boxplot.

# 4 Immunotherapy Response

## 4.1 Immunotherapy_Response()

```
Immunotherapy_Response(gene = 'CD274',SE = MEL_GSE78220)
```

parameter gene is the gene which you wanted. parameter SE is an SummarizedExperiment(SE) object or a list consists of SE objects. The colData of SE objects must contain treatment information names Treatment.

Immunotherapy_Response() perform differential expression analysis and survival analysis in certain gene and return the result.

The score of differential expression analysis is calculated by âˆ’ð‘†ð¼ðºð‘(ð‘™ð‘œð‘”2(ð¹ð¶)) Ã— ð‘™ð‘œð‘”10(ð‘ƒ), where ð¹ð¶ represents the fold change and ð‘ƒ represents the P value derived from the Wilcoxon rank-sum test

The score of survival analysis is calculated by âˆ’ð‘†ð¼ðºð‘(ð‘™ð‘œð‘”2(ð»ð‘…)) Ã— ð‘™ð‘œð‘”10(ð‘ƒ), where ð»ð‘… represents the hazard ratio and ð‘ƒ represents the P value derived from univariate Cox regression analysis.

## 4.2 Visualization

### 4.2.1 plt_diff()

plot differential analysis result between Pre-Treatment and Post-Treatment patients

```
plt_diff(gene = 'CD274', SE = MEL_GSE78220, type = 'Treatment') 
```

plot differential analysis result between Responders and Non-Responders

```
plt_diff(gene = 'CD274', SE = MEL_GSE78220, type = 'Response')
```

### 4.2.2 plt_surv()

plot survival analysis result

```
plt_surv(gene = 'CD274', SE = MEL_GSE78220)
```
