<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 ğŸ“¥ Load data into R environment | tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package</title>
  <meta name="description" content="Chapter 1 ğŸ“¥ Load data into R environment | tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 ğŸ“¥ Load data into R environment | tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 ğŸ“¥ Load data into R environment | tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package" />
  
  
  

<meta name="author" content="Yihao Chen, Li-Na He, Yuanzhe Zhang, Yuelong Shu, Di Zhang, Guangchuang Yu, Zhixiang Zuo" />


<meta name="date" content="2024-05-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="biomarker-evaluation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">tigeR documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>ğŸ“– <strong>Getting start with tigeR</strong></a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>0.1</b> ğŸ”¬ Introduction</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#the-workflow-of-tiger"><i class="fa fa-check"></i><b>0.2</b> ğŸ The workflow of tigeR</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>0.3</b> ğŸ› ï¸ Installation</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#function-overview"><i class="fa fa-check"></i><b>0.4</b> ğŸ—ºï¸ Function Overview</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#troubleshooting"><i class="fa fa-check"></i><b>0.5</b> ğŸ’¥ Troubleshooting</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html"><i class="fa fa-check"></i><b>1</b> ğŸ“¥ Load data into R environment</a>
<ul>
<li class="chapter" data-level="1.1" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html#organization-of-built-in-data"><i class="fa fa-check"></i><b>1.1</b> Organization of built-in data</a></li>
<li class="chapter" data-level="1.2" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html#obtain-data-from-tiger-web-server"><i class="fa fa-check"></i><b>1.2</b> Obtain data from tigeR web server</a></li>
<li class="chapter" data-level="1.3" data-path="load-data-into-r-environment.html"><a href="load-data-into-r-environment.html#obtain-data-from-experimenthub"><i class="fa fa-check"></i><b>1.3</b> Obtain data from ExperimentHub</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html"><i class="fa fa-check"></i><b>2</b> ğŸš© Biomarker Evaluation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#integrate-analysis"><i class="fa fa-check"></i><b>2.1</b> Integrate analysis</a></li>
<li class="chapter" data-level="2.2" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#differential-analysis"><i class="fa fa-check"></i><b>2.2</b> Differential analysis</a></li>
<li class="chapter" data-level="2.3" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#suvival-analysis"><i class="fa fa-check"></i><b>2.3</b> Suvival analysis</a></li>
<li class="chapter" data-level="2.4" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#calculate-comprehensive-signature-score"><i class="fa fa-check"></i><b>2.4</b> Calculate comprehensive signature score</a></li>
<li class="chapter" data-level="2.5" data-path="biomarker-evaluation.html"><a href="biomarker-evaluation.html#assess-the-performance-of-signature"><i class="fa fa-check"></i><b>2.5</b> Assess the Performance of Signature</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumor-microenviroment-analysis.html"><a href="tumor-microenviroment-analysis.html"><i class="fa fa-check"></i><b>3</b> ğŸŒ± Tumor Microenviroment Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="tumor-microenviroment-analysis.html"><a href="tumor-microenviroment-analysis.html#availiable-tme-analysis-method-in-tiger"><i class="fa fa-check"></i><b>3.1</b> Availiable TME Analysis Method in tigeR</a></li>
<li class="chapter" data-level="3.2" data-path="tumor-microenviroment-analysis.html"><a href="tumor-microenviroment-analysis.html#derive-the-proportions-of-different-tme-cell-types"><i class="fa fa-check"></i><b>3.2</b> Derive the Proportions of Different TME Cell Types</a></li>
<li class="chapter" data-level="3.3" data-path="tumor-microenviroment-analysis.html"><a href="tumor-microenviroment-analysis.html#visualization-and-comparing-the-cell-proportions"><i class="fa fa-check"></i><b>3.3</b> Visualization and Comparing the Cell Proportions</a></li>
<li class="chapter" data-level="3.4" data-path="tumor-microenviroment-analysis.html"><a href="tumor-microenviroment-analysis.html#searching-for-key-cell-types-influencing-immune-therapy-response"><i class="fa fa-check"></i><b>3.4</b> Searching for Key Cell Types Influencing Immune Therapy Response</a></li>
<li class="chapter" data-level="" data-path="tumor-microenviroment-analysis.html"><a href="tumor-microenviroment-analysis.html#more-details-about-tme-analysis"><i class="fa fa-check"></i>ğŸ“ More Details about TME Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="prediction-model-construction.html"><a href="prediction-model-construction.html"><i class="fa fa-check"></i><b>4</b> ğŸ¤– Prediction Model Construction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="prediction-model-construction.html"><a href="prediction-model-construction.html#build-model"><i class="fa fa-check"></i><b>4.1</b> Build Model</a></li>
<li class="chapter" data-level="4.2" data-path="prediction-model-construction.html"><a href="prediction-model-construction.html#test-model"><i class="fa fa-check"></i><b>4.2</b> Test Model</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="response-prediction.html"><a href="response-prediction.html"><i class="fa fa-check"></i><b>5</b> ğŸ”® Response Prediction</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="load-data-into-r-environment" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> ğŸ“¥ Load data into R environment<a href="#load-data-into-r-environment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="organization-of-built-in-data" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Organization of built-in data<a href="load-data-into-r-environment.html#organization-of-built-in-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The built-in data follows several specifications for constructing:</p>
<p>â‘  They are constructed into a SummarizedExperiment object using the <strong>SummarizedExperiment()</strong> function in SummarizedExperiment package.</p>
<p>â‘¡ The assays of the SummarizedExperiment object are stored as a SimpleList containing an expression matrix, where rows represent genes and columns represent patients.</p>
<p>â‘¢ The colData of the SummarizedExperiment object is a table(DFrame object of S4), where the row names must be the same as the column names of the expression matrix. The column names of the table represent the clinical information of patients. The detailed information about the column names is presented below.</p>
<div style="width:780px; height:400px; overflow-y: scroll; overflow-x: hidden;">
<table>
<colgroup>
<col width="17%" />
<col width="17%" />
<col width="47%" />
<col width="17%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">column name</th>
<th align="center">Recommend value</th>
<th align="center">Description</th>
<th align="center">Necessity in tigeR analysis</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">sample id</td>
<td align="center">character</td>
<td align="center">The sample identification.</td>
<td align="center">âœ•</td>
</tr>
<tr class="even">
<td align="center">patient_name</td>
<td align="center">character</td>
<td align="center">The patient identification.</td>
<td align="center">âœ•</td>
</tr>
<tr class="odd">
<td align="center">dataset_id</td>
<td align="center">character</td>
<td align="center">The dataset identification.</td>
<td align="center">âœ•</td>
</tr>
<tr class="even">
<td align="center">dataset_group</td>
<td align="center">character</td>
<td align="center"></td>
<td align="center">âœ•</td>
</tr>
<tr class="odd">
<td align="center">Treatment</td>
<td align="center">PRE/POST</td>
<td align="center">The patient sample is collected after Treatment or before Treatment</td>
<td align="center">âœ“</td>
</tr>
<tr class="even">
<td align="center">response</td>
<td align="center">CR/MR/PR/PD/SD/NR/R/N</td>
<td align="center">The Immunotherapy response.</td>
<td align="center">âœ•</td>
</tr>
<tr class="odd">
<td align="center">response_NR</td>
<td align="center">R/N</td>
<td align="center">The Immunotherapy response which only contains R or N.</td>
<td align="center">âœ“</td>
</tr>
<tr class="even">
<td align="center">M.stage</td>
<td align="center">character</td>
<td align="center">The M stage of patients.</td>
<td align="center">âœ•</td>
</tr>
<tr class="odd">
<td align="center">overall.survival..days.</td>
<td align="center">value</td>
<td align="center">The number of days, months or years of survival time of patients.(all samples must be the same unit)</td>
<td align="center">âœ“</td>
</tr>
<tr class="even">
<td align="center">vital.status</td>
<td align="center">Alive/Dead</td>
<td align="center">The survival status of patient.</td>
<td align="center">âœ“</td>
</tr>
<tr class="odd">
<td align="center">Total.Mutation</td>
<td align="center">numeric</td>
<td align="center">The total mutation gene numbers.</td>
<td align="center">âœ•</td>
</tr>
<tr class="even">
<td align="center">Gender</td>
<td align="center">M/F</td>
<td align="center">The gender of patient.</td>
<td align="center">âœ•</td>
</tr>
<tr class="odd">
<td align="center">Therapy</td>
<td align="center">character</td>
<td align="center">The anti-tumor therapy on patients.</td>
<td align="center">âœ•</td>
</tr>
<tr class="even">
<td align="center">age_start</td>
<td align="center">numeric</td>
<td align="center">When the tumor is diagnosed.</td>
<td align="center">âœ•</td>
</tr>
<tr class="odd">
<td align="center">tumor_type</td>
<td align="center">character</td>
<td align="center">The type of tumor.</td>
<td align="center">âœ•</td>
</tr>
<tr class="even">
<td align="center">seq_type</td>
<td align="center">character</td>
<td align="center">The sequencing type.</td>
<td align="center">âœ•</td>
</tr>
<tr class="odd">
<td align="center">id</td>
<td align="center">character</td>
<td align="center">The identification of dataset.</td>
<td align="center">âœ•</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="obtain-data-from-tiger-web-server" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Obtain data from tigeR web server<a href="load-data-into-r-environment.html#obtain-data-from-tiger-web-server" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>Dataloader(c(1,2,3), use_source=&quot;Web Server&quot;)</code></pre>
</div>
<div id="obtain-data-from-experimenthub" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Obtain data from ExperimentHub<a href="load-data-into-r-environment.html#obtain-data-from-experimenthub" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>Dataloader(c(1,2,3), use_source=&quot;ExperimentHub)</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="biomarker-evaluation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-intro.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
