# ðŸŒ± Tumor Microenviroment Analysis

## Availiable TME Analysis Method in tigeR

10 open source tumor microenvironment deconvolution methods are indlcued in tigeR.

::: {style="width:780px; height:200px; overflow-y: scroll; overflow-x: hidden;"}
|                                 Algorithm                                  |                                                    license                                                    |                         PMID                          |  allow custom signature matrix  |
|:--------------------:|:------------------------------:|:----------------:|:--------------:|
|                    [TIMER](http://cistrome.org/TIMER/)                     |                                                free (GPL 2.0)                                                 | [27549193](https://pubmed.ncbi.nlm.nih.gov/27549193/) | âˆš |
|                [CIBERSORT](https://cibersort.stanford.edu/)                |                                       free for non-commercial use only                                        | [25822800](https://pubmed.ncbi.nlm.nih.gov/25822800/) | âˆš |
|             [MCPCounter](https://github.com/ebecht/MCPcounter)             |               free ([GPL 3.0](https://github.com/ebecht/MCPcounter/blob/master/Source/License))               | [27765066](https://pubmed.ncbi.nlm.nih.gov/27765066/) | Ã— |
|                      [xCell](http://xcell.ucsf.edu/)                       |                  free ([GPL 3.0](https://github.com/dviraran/xCell/blob/master/DESCRIPTION))                  | [29141660](https://pubmed.ncbi.nlm.nih.gov/29141660/) | Ã— |
|             [IPS](https://github.com/icbi-lab/Immunophenogram)             |                                                  free (BSD)                                                   | [28052254](https://pubmed.ncbi.nlm.nih.gov/28052254/) | Ã— |
|             [EPIC](https://gfellerlab.shinyapps.io/EPIC_1-1/)              | free for non-commercial use only ([Academic License](https://github.com/GfellerLab/EPIC/blob/master/LICENSE)) | [29130882](https://pubmed.ncbi.nlm.nih.gov/29130882/) | âˆš |
|           [ESTIMATE](https://gfellerlab.shinyapps.io/EPIC_1-1/)            |               free ([GPL 2.0](https://bioinformatics.mdanderson.org/public-software/estimate/))               | [24113773](https://pubmed.ncbi.nlm.nih.gov/24113773/) | Ã— |
|              [ABIS](https://giannimonaco.shinyapps.io/ABIS/)               |                            free ([GPL 2.0](https://github.com/giannimonaco/ABIS))                             | [30726743](https://pubmed.ncbi.nlm.nih.gov/30726743/) | âˆš |
| [ConsensusTME](https://olliecast.shinyapps.io/Deconvolution_Benchmarking/) |              free ([GPL 3.0](https://github.com/cansysbio/ConsensusTME/blob/master/LICENSE.md))               | [31641033](https://pubmed.ncbi.nlm.nih.gov/31641033/) | Ã— |
|       [quanTIseq](http://icbi.at/software/quantiseq/doc/index.html)        |                                                  free (BSD)                                                   | [31126321](https://pubmed.ncbi.nlm.nih.gov/31126321/) | Ã— |
:::

## Derive the Proportions of Different TME Cell Types

â€ƒYou can use the function **deconv_TME()** to derive the proportions of different tumor microenvironment cell types from gene expression data with these tools.
```{r include=FALSE}
library(tigeR)
library(tigeR.data)
library(SummarizedExperiment)
```

```
devtools::install_github('dviraran/xCell')
devtools::install_github("GfellerLab/EPIC")
devtools::install_github("cansysbio/ConsensusTME")
devtools::install_github("federicomarini/quantiseqr")
```
::: {style="width:780px; height:200px; overflow-y: scroll; overflow-x: hidden;"}
```{r echo=TRUE, message=FALSE, warning=FALSE}
## TIMER
frac1 <- deconv_TME(MEL_GSE78220,method="TIMER")

## CIBERSORT
frac2 <- deconv_TME(MEL_GSE78220,method="CIBERSORT")

## MCPCounter
frac3 <- deconv_TME(MEL_GSE78220,method="MCPCounter")

## xCell
frac4 <- deconv_TME(MEL_GSE78220,method="xCell")

## IPS
frac5 <- deconv_TME(MEL_GSE78220,method="IPS")

## EPIC
frac6 <- deconv_TME(MEL_GSE78220,method="epic")

## ESTIMATE
frac7 <- deconv_TME(MEL_GSE78220,method="ESTIMATE")

## ABIS
frac8 <- deconv_TME(MEL_GSE78220,method="ABIS")

## ConsensusTME
frac9 <- deconv_TME(MEL_GSE78220,method="ConsensusTME")

## quanTIseq
frac10 <- deconv_TME(MEL_GSE78220,method="quanTIseq")
```
:::

## Visualization and Comparing the Cell Proportions

```{r echo=TRUE, fig.width=26}
## TIMER
frac1 <- deconv_TME(MEL_GSE78220,method="TIMER")

## CIBERSORT
frac2 <- deconv_TME(MEL_GSE78220,method="CIBERSORT")

cell1 <- c("T cells CD4","Neutrophil", "Macrophage","mDCs","B cells", "T cells CD8")
pie1 <- fraction_pie(cell_name_filter(frac1),feature=factor(cell1, levels = cell1))

cell2 <- c("DCs resting", "T cells CD8", "T cells CD4 naive", "Macrophages M2", "Yd T cells", "Monocytes","Mast cells resting", "Neutrophils", "Tregs","B cells naive")
pie2 <- fraction_pie(cell_name_filter(assay(frac2[1:22,])),
                     feature=factor(cell2, levels = cell2))
pie1
pie2
```

## Searching for Key Cell Types Associated with Immunotherapy Response

```{r echo=TRUE, message=FALSE, warning=FALSE}
## TIMER
TM <- deconv_TME(MEL_GSE91061,method = "TIMER")
TM_SE <- SummarizedExperiment(assays=SimpleList(TM),
                              colData=colData(MEL_GSE91061))
browse_biomk(SE=TM_SE)
```

## Refining a Custom Reference Matrix for TME Deconvolution from Single-cell Sequencing Data
```{r message=FALSE, warning=FALSE}
library(Seurat)
library(tigeR.data)
library(magrittr)
library(GenomicFeatures)
pbmc <- readRDS(system.file("extdata","pbmc.rds",
                            package = "tigeR",
                            mustWork = TRUE))
pbmc <- FindVariableFeatures(pbmc, 
                             selection.method = "vst", 
                             nfeatures = 5000, verbose = FALSE,assay = "RNA")
pbmc <- NormalizeData(pbmc, 
                      normalization.method = "LogNormalize", 
                      scale.factor = 10000, verbose = FALSE)
pbmc <- ScaleData(pbmc, features =  rownames(pbmc), verbose = FALSE)
pbmc <- RunPCA(pbmc, features = VariableFeatures(object = pbmc), verbose = FALSE)
pbmc <- FindNeighbors(pbmc, dims = 1:10, verbose = FALSE)
pbmc <- FindClusters(pbmc, resolution = 0.5, verbose = FALSE)
new.cluster.ids <- c("Naive_CD4T", "CD14_Mono", "Memory_CD4T", "B_cell", "CD8T", "FCGR3A_Mono", "NK", "DC", "Platelet")
names(new.cluster.ids) <- levels(pbmc$seurat_clusters)
pbmc <- RenameIdents(pbmc, new.cluster.ids)
PCAPlot(pbmc, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
pbmc$celltype <- Idents(pbmc)

ref_sig_mtr <- refine_Reference(pbmc, logfc.threshold = 1.5)
dim(ref_sig_mtr)
Ciber_SE <- deconv_TME(MEL_GSE91061,
                       method = "CIBERSORT",
                       sig_matrix=ref_sig_mtr)
diff_TME(Ciber_SE,feature = rownames(Ciber_SE)[1:8])

```

## ðŸ“ More Details About TME Analysis {.unnumbered}

â€ƒ**TIMER** is a comprehensive tool for systematical analysis of immune infiltrates across diverse cancer types.

â€ƒ**CIBERSORT** is an analytical tool from the Alizadeh Lab and Newman Lab to impute gene expression profiles and provide an estimation of the abundances of member cell types in a mixed cell population, using gene expression data.

â€ƒ**MCPCounter** is called the Microenvironment Cell Populations-counter. It allows the robust quantification of the absolute abundance of eight immune and two stromal cell populations in heterogeneous tissues from transcriptomic data

â€ƒ**xCell** is a gene signature-based method learned from thousands of pure cell types from various sources. xCell applies a novel technique for reducing associations between closely related cell types.

â€ƒ**IPS** uses an analytical strategy to provide comprehensive view of 28 TIL subpopulations including effector and memory T cells and immunosuppressive cells (Tregs, MDSCs).

â€ƒ**EPIC** is called Estimating the Proportion of Immune and Cancer cells. It compares the level of expression of genes in a tumor with a library of the gene expression profiles from specific cell types that can be found in tumors and uses this information to predict how many of each type of cell are present.

â€ƒ**ESTIMATE** is described as â€˜Estimation of STromal and Immune cells in MAlignant Tumours using Expression data'. It is a method that utilizes gene expression signatures to infer the proportion of stromal and immune cells within tumor samples.

â€ƒ**ABIS** is called ABsolute Immune Signal (ABIS) deconvolution. ABIS performs absolute deconvolution on RNA-Seq and microarray data.
 
â€ƒ**Consensus**<sup>TME</sup> uses a consensus approch to generate cancer-specific signatres for TME cell types, and utilizes ssGSEA framework to estimate the relative abundance of these cell types.

â€ƒ**quanTIseq** is a computational pipeline for the quantification of the Tumor Immune contexture from human RNA-seq data.
